alertmanager/handlers/handlers.go

```diff
@@ -144,11 +145,13 @@ func tenancyMiddlewareProvider(client client.AlertmanagerClient, getTenantID par
 // receiver and then reloads alertmanager
 func GetReceiverPostHandler(client client.AlertmanagerClient) func(c echo.Context) error {
 	return func(c echo.Context) error {
+		defer glog.Flush()
```

Comment:
Is it necessary to keep flushing?  Doesn't that eliminate the point of a buffer?  Alternatively, if it's low-volume, is there a setting to autoflush?

Reply:
you use defer glog.Flush() to ensure that the logs are printed if the program crashed during the execution of that function. Otherwise the automatic flushing might miss this log (which would be an important one to miss)